cpp/src/arrow/compute/exec/options.h

```diff
@@ -430,6 +430,23 @@ class ARROW_EXPORT SelectKSinkNodeOptions : public SinkNodeOptions {
   SelectKOptions select_k_options;
 };
 
+/// \brief Make a node which selects a range of rows passed through it
+///
+/// All batches pushed to this node will be accumulated, then selected, by the given
+/// fields. Then sorted batches will be forwarded to the generator in sorted order and
+/// select the rows with an offset and a limit.
+class ARROW_EXPORT FetchSinkNodeOptions : public SinkNodeOptions {
```

Comment:
Can we specify sort options (e.g. nulls first or last?)

Reply:
`SortOptions` are now added.